### ISSUE TITLE
Error when calling policy agent on Windows: first path segment in URL cannot contain colon

### ISSUE THREAD

**Issue by jjthom87** [open-policy-agent/opa#2494]
We are running queries with version 0.11.0 and 0.20.5 of open policy agent on Windows, and are getting errors. It seems to be adding a colon somewhere in the line of it getting to the http.send call in the _is_in_org_ function in the rego file below. We believe it's at the substring call in the _is_user_in_org_ in the rego file below. The _is_in_org_ & _is_user_in_org_  functions both trickle up to the _allow_ call that we are attempting to hit. The query to the _allow_ call with identical files work perfectly fine on Mac and Linux.

Using policy agent version 0.20.5

**Call to Policy Agent:**
$ curl -s -X POST -d "{ \"input\": {\"user\": \"moe\", \"action\": \"Create\", \"resource\": \"resource.uat\"} }" localhost:8181/v1/data/my_rule_package/allow?pretty=true

**Response:**
```
{
  "code": "internal_error",
  "message": "error(s) occurred while evaluating query",
  "errors": [
    {
      "code": "eval_builtin_error",
      "message": "http.send: Get https://call-to-pip.net\"/subjects/moe/orgs/ORG: dial tcp: lookup call-to-pip.net\": no such host",
      "location": {
        "file": "\\Users\\path\\to\\orgs.rego",
        "row": 17,
        "col": 2
      }
    }
  ]
}

```
 

 

Using policy agent version 0.11.0
 
**Call to Policy Agent:**
curl -s -X POST -d "{ \"input\": {\"user\": \"moe\", \"action\": \"Create\", \"resource\": \"resource.uat\"} }" localhost:8181/v1/data/my_rule_package/allow?pretty=true

**Response:**
```
{
  "code": "internal_error",
  "message": "error(s) occurred while evaluating query",
  "errors": [
    {
      "code": "eval_builtin_error",
      "message": "http.send: parse \\\"https://call-to-pip.net\\\"/subjects/moe/orgs/ORG: first path segment in URL cannot contain colon",
      "location": {
        "file": "\\Users\\path\\to\\orgs.rego",
        "row": 17,
        "col": 2
      }
    }
  ]
}
```

 

Here are our rego files:

```
Package test.rbac

is_user_in_org(userId, member) {
    startswith(member, "mne:")
    substring(member, 4, -1, mne)
    data.test.orgs.is_in_org(userId, mne)
}
```

```
package test.orgs

is_in_org(comit, org) = true {
    not is_null(url)
    temp_array = [url, "/subjects/", comit, "/orgs/", org]
    concat("",temp_array,formatted_url)
    http.send({"method":"get","url":formatted_url}, output)
    output.body.member = true
}
```

 

Here is an example data.json, the data causing this problem is the member **“mne:ORG”**:
```
{
  "roles": [
    {
      "name": "Stooge",
      "responsibilities": [
        {
          "name": "Monitor DEV",
          "resources": ["resource.dev"],
          "actions": ["Read"]
        }
      ],
      "members": [
        "moe",
        "larry",
        "curly",
        "shemp"
      ]
    },
    {
      "name": "Head Stooge",
      "responsibilities": [
        {
          "name": "Manage DEV and UAT",
          "resources": ["regexp:resource\\.(dev|uat)"],
          "actions": ["Create", "Read", "Update", "Delete"]
        }
      ],
      "members": [
        "moe",
        "mne:ORG"
      ]
    }
  ]
}
```

Also, the error only happens when we are running the policy agent and attempting to hit policy endpoints. The error does not happen when we run our policy agent tests against the _allow_ call. We are sure that the the query is reaching the http.send call as part of the test.

**Tests:**
```
package my_rule_package

test_should_allow_userone_to_create_a_uat_resource {
    allow with input as { "user": "userone", "action": "Create", "resource": "resource.uat" }
}

test_should_not_allow_usertwo_to_create_a_prod_resource {
    not allow with input as { "user": "usertwo", "action": "Create", "resource": "resource.prod" }
}

```
 

**Test Results:**
data.my_rule_package.test_should_allow_userone_to_create_a_uat_resource: PASS (153.0127ms)
data.my_rule_package.test_should_not_allow_usertwo_to_create_a_prod_resource: PASS (999.3?µs)

**Comment by patrick-east** [comment]
Can you share a little bit more info about the policy? I tried to reproduce this and was unable to.

I took your snippets and made a couple of quick modifications:

t1.rego:
```rego
package test.rbac

is_user_in_org(userId, member) {
    startswith(member, "mne:")
    substring(member, 4, -1, mne)
    data.test.orgs.is_in_org(userId, mne)
}
```
t2.rego:
```rego
package test.orgs

url := "http://google.com"

is_in_org(comit, org) = true {
    #not is_null(url)
    temp_array = [url, "/subjects/", comit, "/orgs/", org]
    concat("",temp_array,formatted_url)
    trace(sprintf("formatted_url = %s", [formatted_url]))
    http.send({"method":"get","url":formatted_url}, output)
    output.body.member = true
}
```

The key changes being that I define a `url`, ignore the `not is_null(url)` thing, and added in a `trace()` to see what the formatted url looks like.

On my windows box:

```
PS C:\Users\patrick> .\opa.exe run C:\projects\issue\2494\t1.rego C:\projects\issue\2494\t2.rego
OPA 0.21.0 (commit 3807c481, built at 2020-06-16T15:40:14Z)

Run 'help' to see a list of commands and check for updates.

> notes
> data.test.rbac.is_user_in_org("123", "mne:ORG")
query:1                            Enter data.test.rbac.is_user_in_org("123", "mne:ORG", _)
\projects\issue\2494\t1.rego:3     | Enter data.test.rbac.is_user_in_org
\projects\issue\2494\t2.rego:5     | | Enter data.test.orgs.is_in_org
\projects\issue\2494\t2.rego:9     | | | Note "formatted_url = http://google.com/subjects/123/orgs/ORG"
undefined
>
```

Where there is no error and the formatted url looks pretty good.

If you can please share some samples that reproduce the issue. In particular information about what the `url` value is that you end up using would be helpful. I see in the `0.10.0` error you are getting  `dial tcp: lookup call-to-pip.net\": no such host` which seems more like a dns issue, the next one mentions `first path segment in URL cannot contain colon` was that with exactly the same url string?

**Comment by jjthom87** [comment]
So, we have figured out the problem here. The problem, which seems to be with Windows, was that we had quotes around our Urls, which we were pulling into the rego file as environment variables. 

On this line here, the url is being pulled in via the env var
`temp_array = [url, "/subjects/", comit, "/orgs/", org]`

We had the env var set as: `"https://call-to-pip.net"`
We changed the env var remove the quotes `https://call-to-pip.net`

This fixed the problem for both versions of OPA that we were testing.
